recipe/build.sh

```diff
@@ -1,12 +1,24 @@
 #!/bin/bash
 
-cp -r ${BUILD_PREFIX}/share/libtool/build-aux/config.* .
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* . || true
 
-export LD_LIBRARY_PATH=$PREFIX/lib:$LD_LIBRARY_PATH
+if [[ "$target_platform" == "win-64" ]]; then
+  export PREFIX=${PREFIX}/Library
+fi
 
 ./configure --prefix=$PREFIX \
             --with-gmp=$PREFIX \
-            --enable-static
-make
-make check
+            --disable-static \
+            --enable-thread-safe
+
+make -j${CPU_COUNT}
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != 1 && "${CROSSCOMPILING_EMULATOR}" == "" ]]; then
+  make check
+fi
 make install
+
+if [[ "$target_platform" == "win-64" ]]; then
+  cp ${PREFIX}/lib/libmpfr.dll.a ${PREFIX}/lib/mpfr.lib
```

Comment:
Is this valid? An MSVC import library (`.lib`) is not the same as an import library produced by GCC/LLVM (`dll.a`). I'm also not sure if it's valid to rename it from `libmpfr` to `mpfr`. Same below with the dll. Usually the import lib file has the dll name hardcoded in it.

Reply:
I think you're right. The .dll is copied, though, not renamed. So I think that bit is ok.
So unless we generate a .lib file properly from the .dll using lib.exe, all downstreams need to use msys2 as well. At first glance it looks like most of the downstreams will use msys2 (r packages, other similar maths libraries like mpc) but I think it'd be good to solve this problem properly now

Reply:
I think the best way to see if renaming is even necessary (for the `.dll.a`) would be to test locally. https://gcc.gnu.org/onlinedocs/gnat_ugn/Introduction-to-Dynamic-Link-Libraries-DLLs.html could help.

Worst case scenario, if the DLL (not the `.ddl.a`) if really using the MSVC's runtime, then we could generate a `.lib` outself. We already do that in a couple of places.

Reply:
I've been having a look at generating a .lib. However, as currently configured, this recipe will generate a library which depends on `gmp`:
`./Library/include/mpfr.h:#include <gmp.h>`
and currently there's only an msys2-gmp and msys2-gmp-devel, not an msvc-based gmp.
Will that be an issue for libraries linking to mpfr? Maybe not.

Reply:
I think I also need to add gmp at runtime since there's no run_exports in [msys2-gmp](https://github.com/AnacondaRecipes/repack_msys2/blob/main/msys2-gmp-feedstock/recipe/meta.yaml). @ifitchet I guess this is the case for many msys2 recipes? Is there a reason it's not needed?

Reply:
I tried to compile `gmp` with MSVC but dropped that idea https://github.com/AnacondaRecipes/gmp-feedstock/pull/4

Reply:
> I think I also need to add gmp at runtime since there's no run_exports in [ucrt64-gmp](https://github.com/AnacondaRecipes/repack_msys2/blob/main/mingw-w64-gmp-feedstock/recipe/meta.yaml). @ifitchet I guess this is the case for many msys2 recipes? Is there a reason it's not needed?

To consolidate the data points: There are no `run_exports` across MSYS2 old or new -- or conda-forge, I suspect.

We didn't want to change too many variables at once.  The new MSYS2 should operate like the old MSYS2 (`m2-*`, `m2w64-*`) with most of its foibles.

There's nothing to stop us revisiting -- although I'm not sure how we might encode the (relatively random) sprinkling of `run-exports` across 160 automatically generated recipes.  A YAML file, no doubt.

There are interesting features ahead as the "pypi cran skeleton" has, not only `m2-*` and `m2w64-*` hard-coded, but also hard-codes `{{ native }}gcc-libs  # [win-64]` in `requirements.run`.  I did write some variants code but the pypi skeletons are deprecated?  The `gcc-libs` bit would obviously be replaced by `run-exports` functionality.